pre_commit/commands/run.py

```diff
@@ -417,10 +424,68 @@ def run(
         to_install = [hook for hook in hooks if hook.id not in skips]
         install_hook_envs(to_install, store)
 
-        retval = _run_hooks(config, hooks, skips, args, environ)
+        if args.hook_stage == 'commit' and not is_git_message_supplied():
+            # Allow user to enter commit message concurrently with running pre-commit hooks to lower
+            # wait times.
+
+            # Run git commands before starting hooks to avoid race conditions and git lock errors.
+            commit_message_template = get_commit_message_template()
+
+            with concurrent.futures.ThreadPoolExecutor(max_workers=2) as ex:
```

Comment:
Is this current `max_workers` value temporary? Just curious behind the logic for picking 2.

Reply:
idk, we're only running two things; we only need two threads 🤷‍♀️